files/default/neutron-ha-tool.py

```diff
@@ -747,28 +753,39 @@ def migrate_router(qclient, router, agent, target,
         nscleanup.delete_router_namespace(router['id'])
 
 
-def wait_router_migrated(qclient, router_id, target_host, maxtries=60):
+def wait_router_migrated(qclient, router_id, target_host, maxtries=60,
+                         state="ACTIVE"):
     """
     Returns nothing. Waits for all non-distributed ports and floating IPs
-    of a router for being in the ACTIVE again after a migration.
+    of a router for reaching a specific state (ACTIVE by default) during
+    a migration.
 
     :param qclient: A neutron client
     :param router_id: The id of the router to check
-    :param target_host: The host on which the ports should be ACTIVE
-                        (reflected by the binding:host_id attribute of
-                        a port)
+    :param target_host: The host on which the ports should reach the requested
+                        state. Reflected by the binding:host_id attribute of
+                        a port.
     :param maxtries: Maximum number of times the ports' status is polled,
-                     an expection is raised there are still ports with a
-                     non ACTIVE status after that.
+                     an expection is raised if there are still ports with a
+                     in state that doesn't match the expectation.
+    :param state: The Port state to wait for, can be "ACTIVE" or "DOWN"
     """
 
     LOG.info("Wait for the ports and floating IPs of router_id=%s "
-             "to be ACTIVE", router_id)
+             "to be %s", router_id, state)
     remaining_ports = ["dummy"]
-    remaining_fips = ["dummy"]
+    # For the "DOWN" case it is enough to check for the router ports
+    # as the floating IPs don't change their state until re-assigned
+    remaining_fips = [] if state == 'DOWN' else ['dummy']
+
     remaining_iterations = maxtries
     while remaining_iterations:
         if remaining_ports:
+            expected_host_id = [target_host]
```

Comment:
That's a nitpicking, but shouldn't it be `expected_host_ids`?

Reply:
True. And fixed.